diff --git a/node_modules/algosdk/dist/cjs/encoding/schema/bytearray.js b/node_modules/algosdk/dist/cjs/encoding/schema/bytearray.js
index 99f77ca..15fb232 100644
--- a/node_modules/algosdk/dist/cjs/encoding/schema/bytearray.js
+++ b/node_modules/algosdk/dist/cjs/encoding/schema/bytearray.js
@@ -34,6 +34,9 @@ class ByteArraySchema extends encoding_js_1.Schema {
         throw new Error(`Invalid byte array: (${typeof data}) ${data}`);
     }
     fromPreparedJSON(encoded) {
+        if (encoded === null || encoded === undefined) {
+          return new Uint8Array();
+        }
         if (typeof encoded === 'string') {
             return (0, binarydata_js_1.base64ToBytes)(encoded);
         }
diff --git a/node_modules/algosdk/dist/esm/encoding/schema/bytearray.js b/node_modules/algosdk/dist/esm/encoding/schema/bytearray.js
index d9ed6fa..5bd9eb5 100644
--- a/node_modules/algosdk/dist/esm/encoding/schema/bytearray.js
+++ b/node_modules/algosdk/dist/esm/encoding/schema/bytearray.js
@@ -31,6 +31,9 @@ export class ByteArraySchema extends Schema {
         throw new Error(`Invalid byte array: (${typeof data}) ${data}`);
     }
     fromPreparedJSON(encoded) {
+        if (encoded === null || encoded === undefined) {
+          return new Uint8Array();
+        }
         if (typeof encoded === 'string') {
             return base64ToBytes(encoded);
         }
diff --git a/node_modules/algosdk/src/encoding/schema/bytearray.ts b/node_modules/algosdk/src/encoding/schema/bytearray.ts
index 5218049..6648087 100644
--- a/node_modules/algosdk/src/encoding/schema/bytearray.ts
+++ b/node_modules/algosdk/src/encoding/schema/bytearray.ts
@@ -48,6 +48,9 @@ export class ByteArraySchema extends Schema {
   }
 
   public fromPreparedJSON(encoded: JSONEncodingData): Uint8Array {
+    if (encoded === null || encoded === undefined) {
+      return new Uint8Array();
+    }
     if (typeof encoded === 'string') {
       return base64ToBytes(encoded);
     }
